### ISSUE TITLE
Typechecking of first cypress test hangs

### ISSUE THREAD

**Issue by wvanderdeijl** [nrwl/nx#2063]
## Current Behavior

Since upgrading `@nrwl/cypress` from 8.4.13 to 8.7.0 our first test fails to load. This does not happen when running in watch mode. With debugging enabled (`DEBUG=*` env var) we can see that `cypress:webpack` triggers the type checking service.
But type-checking for the first spec file never completes. After a couple of minutes the started cypress runner gives up and reports the first spec file as green/okay, but with 0 tests completed. It then continues with the other spec files and they run fine.

After some debugging we feel this was introduced by #1872
Our project has a custom support file configured in the cypress.json configuration. Looking at the debugging we see that both the initial spec file and the `support/index` file are requested when the runner starts. The preprocessor first starts working on the first spec file and spawns the type checker. But before the type checking is completed for the first spec file, the preprocessor is triggered for the `support/index` file. But the `fork-ts-checker-webpack-plugin` module seems to support only a single concurrent request and it will cancel the running typechecking request for the spec file and starts working on the `support/index` file. This means the type checking for the first spec file never completes, and thus the preprocessor for the first spec file never completes.

Debugging has indicated the first typechecking request is cancelled at https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/blob/032475e8d58a2b9d55c70285c9778cae2ce2aa96/src/index.ts#L476

As a very hacky workaround I have forced the `@nrwl/cypress` preprocessor to only work on a single file at a time:
```
module.exports = (on: Cypress.Actions, config: Cypress.ConfigOptions) => {
    const processor = preprocessTypescript(config);
    let pending = Promise.resolve();
    on('file:preprocessor', async (...args: any[]) => {
        await pending;
        pending = processor(...args);
        return pending;
    });
    return config;
};

```

### Failure Logs

Here is the relevant snippet from the console when running with `DEBUG=*`
```

  cypress:server:preprocessor getFile /Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/integration/auth.spec.ts +0ms
  cypress:server:appdata path: /Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/integration/auth.spec.ts +1s
  cypress:server:plugins plugin event registered? { event: 'file:preprocessor', isRegistered: true } +1s
  cypress:server:plugins execute plugin event 'file:preprocessor' with args: EventEmitter { domain: null, _events: { rerun: [Function] }, _eventsCount: 1, _maxListeners: undefined, filePath: '/Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/integration/auth.spec.ts', shouldWatch: false, outputPath: '/Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/integration/auth.spec.ts' } undefined undefined +0ms
  cypress:server:plugins call event file:preprocessor for invocation id inv1 +1ms
  cypress:server:plugins:child execute plugin event: file:preprocessor ({ eventId: 2, invocationId: 'inv1' }) +2s
  cypress:webpack get /Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/integration/auth.spec.ts +2s
  cypress:webpack webpackOptions: { resolve: { extensions: [ '.ts', '.tsx', '.mjs', '.js', '.jsx' ], plugins: [ [Object] ] }, module: { rules: [ [Object] ] }, plugins: [ ForkTsCheckerWebpackPlugin { options: [Object], tsconfig: '/Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/tsconfig.e2e.json', tslint: undefined, watch: [], ignoreDiagnostics: [], ignoreLints: [], logger: [Object], silent: false, async: true, checkSyntacticErrors: false, workersNumber: 1, memoryLimit: 2048, useColors: true, colors: [Object], formatter: [Function: defaultFormatter], tsconfigPath: undefined, tslintPath: undefined, watchPaths: [], compiler: undefined, started: undefined, elapsed: undefined, cancellationToken: undefined, isWatching: false, checkDone: false, compilationDone: false, diagnostics: [], lints: [], emitCallback: [Function: noopEmitCallback], doneCallback: [Function: doneCallback], typescriptVersion: '3.5.3', tslintVersion: undefined, vue: false } ], externals: [ [Function] ] } +0ms
  cypress:webpack watchOptions: {} +0ms
  cypress:webpack input: /Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/integration/auth.spec.ts +1ms
  cypress:webpack output: /Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/integration/auth.spec.ts +0ms
  compression no compression: size below threshold +178ms
  express:router dispatching GET /__cypress/tests?p=src/support/index.js-979 +26ms
  express:router query  : /__cypress/tests?p=src/support/index.js-979 +0ms
  express:router expressInit  : /__cypress/tests?p=src/support/index.js-979 +0ms
  express:router <anonymous>  : /__cypress/tests?p=src/support/index.js-979 +0ms
  express:router cookieParser  : /__cypress/tests?p=src/support/index.js-979 +0ms
  express:router compression  : /__cypress/tests?p=src/support/index.js-979 +0ms
  express:router errorHandler  : /__cypress/tests?p=src/support/index.js-979 +1ms
  cypress:server:controllers:spec request for { spec: 'src/support/index.js' } +0ms
  cypress:server:preprocessor getFile /Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/support/index.js +25ms
  cypress:server:appdata path: /Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/support/index.js +25ms
  cypress:server:plugins plugin event registered? { event: 'file:preprocessor', isRegistered: true } +24ms
  cypress:server:plugins execute plugin event 'file:preprocessor' with args: EventEmitter { domain: null, _events: { rerun: [Function] }, _eventsCount: 1, _maxListeners: undefined, filePath: '/Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/support/index.js', shouldWatch: false, outputPath: '/Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/support/index.js' } undefined undefined +0ms
  cypress:server:plugins call event file:preprocessor for invocation id inv2 +1ms
  express:router dispatching GET /__cypress/tests?p=src/integration/auth.spec.ts-790 +1ms
  express:router query  : /__cypress/tests?p=src/integration/auth.spec.ts-790 +0ms
  express:router expressInit  : /__cypress/tests?p=src/integration/auth.spec.ts-790 +0ms
  express:router <anonymous>  : /__cypress/tests?p=src/integration/auth.spec.ts-790 +0ms
  express:router cookieParser  : /__cypress/tests?p=src/integration/auth.spec.ts-790 +0ms
  express:router compression  : /__cypress/tests?p=src/integration/auth.spec.ts-790 +0ms
  express:router errorHandler  : /__cypress/tests?p=src/integration/auth.spec.ts-790 +0ms
  cypress:server:controllers:spec request for { spec: 'src/integration/auth.spec.ts' } +1ms
  cypress:server:preprocessor getFile /Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/integration/auth.spec.ts +1ms
  cypress:server:plugins plugin event registered? { event: 'file:preprocessor', isRegistered: true } +1ms
  cypress:server:preprocessor headless and already processed +1ms
Starting type checking service...
Using 1 worker with 2048MB memory limit
  cypress:server:plugins:child execute plugin event: file:preprocessor ({ eventId: 2, invocationId: 'inv2' }) +689ms
  cypress:webpack get /Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/support/index.js +686ms
  cypress:webpack webpackOptions: { resolve: { extensions: [ '.ts', '.tsx', '.mjs', '.js', '.jsx' ], plugins: [ [Object] ] }, module: { rules: [ [Object] ] }, plugins: [ ForkTsCheckerWebpackPlugin { options: [Object], tsconfig: '/Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/tsconfig.e2e.json', tslint: undefined, watch: [], ignoreDiagnostics: [], ignoreLints: [], logger: [Object], silent: false, async: true, checkSyntacticErrors: false, workersNumber: 1, memoryLimit: 2048, useColors: true, colors: [Object], formatter: [Function: defaultFormatter], tsconfigPath: '/Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/tsconfig.e2e.json', tslintPath: null, watchPaths: [], compiler: [Object], started: [Array], elapsed: undefined, cancellationToken: [Object], isWatching: false, checkDone: false, compilationDone: false, diagnostics: [], lints: [], emitCallback: [Function: noopEmitCallback], doneCallback: [Function: doneCallback], typescriptVersion: '3.5.3', tslintVersion: undefined, vue: false, service: [Object] } ], externals: [ [Function] ], entry: [ '/Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/integration/auth.spec.ts' ], output: { path: '/Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/integration', filename: 'auth.spec.ts' }, devtool: 'inline-source-map' } +0ms
  cypress:webpack watchOptions: {} +1ms
  cypress:webpack input: /Users/xxxxx/git/xxxxx/apps/xxxxx-e2e/src/support/index.js +0ms
  cypress:webpack output: /Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/support/index.js +0ms
  cypress:webpack warnings for /Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/support/index.js +4s
  cypress:webpack [ 'configuration\nThe \'mode\' option has not been set, webpack will fallback to \'production\' for this value. Set \'mode\' option to \'development\' or \'production\' to enable defaults for each environment.\nYou can also set it to \'none\' to disable any default behavior. Learn more: https://webpack.js.org/configuration/mode/' ] +0ms
  cypress:webpack finished bundling /Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/support/index.js +0ms
  cypress:server:plugins promise resolved for id 'inv2' with value /Users/xxxxx/Library/Application Support/Cypress/cy/production/projects/xxxxx-e2e-707925767bd1fa3dd936e806b6a122ce/bundles/src/support/index.js +0ms
```

## Other

There is a good chance this is related to #1976

**Comment by FrozenPandaz** [comment]
Fixed in #1920

**Comment by Janiczek** [comment]
For anybody who needs this: here's a version for `webpack` preprocessor (instead of TS one).

```js
const webpackPreprocessor = require('@cypress/webpack-preprocessor');

module.exports = (on) => {
  let pending = Promise.resolve();
  const webpackOptions = require('../../webpack.config.js');
  const preprocessor = webpackPreprocessor({webpackOptions});
  on('file:preprocessor', async (...args) => {
    await pending;
    pending = preprocessor(...args);
    return pending;
  });
};
```
